web/ui/src/Hosts/HostDetailsController.js

```diff
@@ -1,36 +1,161 @@
-/* globals controlplane: true */
-
 /* HostDetailsController
  * Displays list of hosts
  */
 (function() {
     'use strict';
 
-    controlplane.controller("HostDetailsController", [
-    "$scope", "$routeParams", "$location", "resourcesFactory", "authService",
-    "$modalService", "$translate", "miscUtils", "hostsFactory", "$notification",
-    "instancesFactory", "servicesFactory", "$interval",
-    function($scope, $routeParams, $location, resourcesFactory, authService,
-    $modalService, $translate, utils, hostsFactory, $notification,
-    instancesFactory, servicesFactory, $interval){
-        // Ensure logged in
-        authService.checkLogin($scope);
-
-        $scope.name = "hostdetails";
-        $scope.params = $routeParams;
-
-        $scope.breadcrumbs = [
-            { label: 'breadcrumb_hosts', url: '/hosts' }
-        ];
-
-        $scope.viewLog = function(instance) {
-            $scope.editService = angular.copy(instance);
-            resourcesFactory.getInstanceLogs(instance.model.ServiceID, instance.id)
+    class HostDetailsController {
+
+        constructor($scope, $routeParams, $location, resourcesFactory, authService,
+                    $modalService, $translate, $notification, $interval, servicedConfig,
+                    log, miscUtils, Host, Instance, $q) {
+
+            authService.checkLogin(this);
+
+            this.resourcesFactory = resourcesFactory;
+            this.$modalService = $modalService;
+            this.$translate = $translate;
+            this.$notification = $notification;
+            this.utils = miscUtils;
+            this.Host = Host;
+            this.params = $routeParams;
+            this.Instance = Instance;
+            this.$interval = $interval;
+            this.$q = $q;
```

Comment:
Please use the pattern described in a comment on `HostsController.js`. to clean up this dependency stuff.